server/services/ProfilesService.js

```diff
@@ -387,27 +375,133 @@ const unblockProfile = function (myId, blockedProfileId) {
+    .limit(followingsLimit)
+    .orderBy('actionDate', 'desc')
+  if (paginationId) {
+    const paginateRef = db.collection('users').doc(profileId)
+    return paginateRef.get().then(snapshot => {
+      const next = followingsQueryRef.startAfter(snapshot)
+      return next.get().then(docSnapshots => {
+        const docsLength = docSnapshots.docs.length
+        if (docsLength > 0) {
+          const lastIndex =
+            docsLength >= followingsLimit
+              ? docSnapshots.docs[docSnapshots.docs.length - 1].id
+              : 'Last index'
+          const followingsProfiles = []
+          docSnapshots.forEach(doc => {
+            profilesRef
+              .doc(doc.id)
+              .get()
+              .then(doc => {
+                followingsProfiles.push(doc.data())
+              })
+          })
+          return Promise.all(followingsProfiles).then(values => {
+            return { data: values, lastIndex }
+          })
+        } else {
+          return 'No more users left'
+        }
       })
-    }
-    const followings = doc.data().followings
-    if (!followings || followings.length === 0) {
-      throw new CustomError({
-        name: 'DatabaseError',
-        message: 'User with given id doesnt have followings',
-        status: 404
+    })
+  } else {
+    return followingsQueryRef.get().then(docSnapshots => {
+      const docsLength = docSnapshots.docs.length
+      if (docsLength > 0) {
+        const lastIndex =
+          docsLength >= followingsLimit
+            ? docSnapshots.docs[docSnapshots.docs.length - 1].id
+            : 'Last query'
+        const followingsProfiles = []
+        docSnapshots.forEach(doc => {
+          followingsProfiles.push(
+            profilesRef
+              .doc(doc.id)
+              .get()
+              .then(doc => {
+                return doc.data()
+              })
+          )
+        })
+        return Promise.all(followingsProfiles).then(values => {
+          return { data: values, lastIndex }
+        })
+      } else {
+        return 'No more users left'
+      }
+    })
+  }
+}
+const getFollowersById = function(profileId, followersLimit, paginationId) {
+  const profilesRef = db.collection('users')
+  const followersQueryRef = db
+    .collection('users')
+    .doc(profileId)
+    .collection('followers')
+    .limit(followersLimit)
+    .orderBy('actionDate', 'desc')
+  if (paginationId) {
+    const paginateRef = db.collection('users').doc(profileId)
+    return paginateRef.get().then(snapshot => {
+      const next = followersQueryRef.startAfter(snapshot)
+      return next.get().then(docSnapshots => {
+        const docsLength = docSnapshots.docs.length
+        if (docsLength > 0) {
+          const lastIndex =
+            docsLength >= followersLimit
+              ? docSnapshots.docs[docSnapshots.docs.length - 1].id
+              : 'Last index'
+          const followersProfiles = []
+          docSnapshots.forEach(doc => {
```

Comment:
> Please make sure you aren't doing extra work(reads to database):
> try checking "docSnapshots.forEach((doc) => console.log(doc.data()))" - if doc already contains required information needed for followers - then there is no more need in "profilesRef.doc(doc.id).get()"
> 
> Im not sure exactly what information docSnapshots really hold but it's worth the try, I guess

Let me explain why i did like this(by the way i changed this logic a bit too, again 71 pull request, but idea keeps the same)
First of all i'll answer on your tip 'Im not sure exactly what information docSnapshots really hold but it's worth the try, I guess'
In that subcollections we store only profile id and a date of a subscription
And probably i need to provide more answers to clearify situation
1) Why i did in that way?
With this data structure we can easily load followers or followings with pagination, and show this data sorted by Date(latest followers of followings shows firstly)
2) But more important question should be why i decided not to store more data like nickname or avatar in that subcollections? - 
In that case we woukld have less read/writes, but at the same moment, each time when user decide to change avatar or nickname, it would need to change this fields directly in each followers/followings document where it exists.
So  at my humble opinion this approach makes logic with backend more simple and predictable and even more economical in long term perspective.